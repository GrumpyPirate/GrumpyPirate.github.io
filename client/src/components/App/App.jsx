// React
import React from 'react'

// Router
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'

// Components
import Sidebar from 'components/Sidebar/Sidebar'
// import MainContainer from 'components/MainContainer/MainContainer'
// import Footer from 'components/Footer/Footer'
// import About from 'components/About/About'
// import ScrollToTop from 'components/ScrollToTop/ScrollToTop'
// import WebDev from 'components/WebDev/WebDev'
// import PortfolioPage from 'components/Portfolio/PortfolioPage/PortfolioPage'
// import HTTPNotFound from 'components/404/404'

// SCSS
import './App.scss'

// Define App
const App = () => (
  <Router basename="/">
    <div className="app">
      <Sidebar />
      {/* <MainContainer> */}
      {/*   <Switch> */}
      {/*     <ScrollToTop> */}
      {/*       <div> */}
      {/*         <Route exact path="/" component={About} /> */}
      {/*         {/* <Route exact path="/webdev" component={WebDev} /> */}
      {/*         {/* <Route path="/webdev/:portfolioItemSlug" component={PortfolioPage} /> */}
      {/*         {/* <Route component={HTTPNotFound} /> */}
      {/*       </div> */}
      {/*     </ScrollToTop> */}
      {/*   </Switch> */}
      {/*   <Footer /> */}
      {/* </MainContainer> */}
    </div>
  </Router>
)

export default App
