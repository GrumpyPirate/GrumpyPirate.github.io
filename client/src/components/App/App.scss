@import 'core';

.app {
  padding-top: $sidebar-height-mobile;

  // lg up
  @include respond-up(lg) {
    padding-top: 0;
    display: flex;
    flex-flow: row nowrap;
    align-items: stretch;
  }

  &__page-transition {
    &--enter {
      position: relative;
      opacity: 1;

      &--active {
        opacity: 1;
        transition-property: opacity;
        transition-duration: $anim-dur * 2;
      }
    }

    &--exit {
      &--active {
        position: absolute;
        left: 0;
        right: 0;
        height: 100%;
        opacity: 0;
        transition-property: opacity;
        transition-duration: $anim-dur * 2;
        z-index: 1;
      }
    }
  }
}
