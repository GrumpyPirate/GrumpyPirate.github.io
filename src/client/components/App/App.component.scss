@import 'core';

.app {
  padding-top: $sidebar-height-mobile;

  @include respond-up('lg') {
    align-items: stretch;
    display: flex;
    flex-flow: row nowrap;
    padding-top: 0;
  }

  &__page-transition {
    &--enter {
      &,
      &-active {
        height: 100%;
        left: 0;
        opacity: 0;
        position: absolute;
        right: 0;
        top: 0;
        z-index: 10;
      }

      &--active {
        opacity: 1;
        transition-duration: $anim-dur * 2;
        transition-property: opacity;
      }
    }

    &--exit {
      &--active {
        opacity: 0;
        pointer-events: none;
        transition-duration: $anim-dur * 2;
        transition-property: opacity;
      }
    }
  }
}
