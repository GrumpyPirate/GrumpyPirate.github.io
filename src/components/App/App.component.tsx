import { LocationState } from 'history';
import React, { FunctionComponent } from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import CSSTransition from 'react-transition-group/CSSTransition';
import TransitionGroup from 'react-transition-group/TransitionGroup';

import HTTPNotFound from 'components/404/404.component';
import About from 'components/About/About.container';
import Footer from 'components/Footer/Footer.component';
import Main from 'components/Main/Main.container';
import Portfolio from 'components/Portfolio/Portfolio.container';
import PortfolioPage from 'components/PortfolioPage/PortfolioPage.container';
import Sidebar from 'components/Sidebar/Sidebar.container';
import animation from 'config/animation';

import classes from './App.component.scss';

const App: FunctionComponent = () => (
  <Router basename="/">
    <div className={classes['app']}>
      <Sidebar />
      <Main
        renderLocation={(location: LocationState) => {
          const { pathname } = location;

          return (
            <TransitionGroup component={null}>
              <CSSTransition
                key={pathname}
                classNames={{
                  enter: classes['app__page-transition--enter'],
                  enterActive: classes['app__page-transition--enter--active'],
                  exit: classes['app__page-transition--exit'],
                  exitActive: classes['app__page-transition--exit--active'],
                  exitDone: classes['app__page-transition--exit--done'],
                }}
                timeout={{ exit: animation.duration * 2, enter: animation.duration * 2 }}
                mountOnEnter
              >
                <Route
                  location={location}
                  render={() => (
                    <Switch>
                      <Route exact path="/" component={About} />
                      <Route exact path="/portfolio" component={Portfolio} />
                      <Route exact path="/portfolio/:slug" component={PortfolioPage} />
                      <Route exact path="/404" component={HTTPNotFound} />
                      <Route component={HTTPNotFound} />
                    </Switch>
                  )}
                />
              </CSSTransition>
            </TransitionGroup>
          );
        }}
      >
        <Footer />
      </Main>
    </div>
  </Router>
);

export default App;
