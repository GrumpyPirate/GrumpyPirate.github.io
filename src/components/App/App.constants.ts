import styled, { createGlobalStyle } from 'styled-components';

import Spinner from 'components/Spinner/Spinner';
import {
  animDur,
  bodyBgDark,
  bodyColorDark,
  bodyFontFamily,
  bodyFontWeight,
  fontWeights,
  lineHeight,
  linkActiveColor,
  linkColor,
  linkFontWeight,
  linkHoverColor,
  mediaQueries,
  noOutline,
  noUnderline,
  palette,
  paragraphMarginBottom,
  rem,
  sidebarHeightMobile,
} from 'styles';

export const GlobalStyles = createGlobalStyle`
  html,
  body {
    background-color: ${bodyBgDark};
  }

  body {
    color: ${bodyColorDark};
    font-family: ${bodyFontFamily};
    font-weight: ${bodyFontWeight};
    line-height: ${lineHeight};
  }

  p {
    margin: 0 0 ${paragraphMarginBottom};
  }

  a {
    ${noOutline}
    ${noUnderline}

    color: ${linkColor};
    font-weight: ${linkFontWeight};
    transition-duration: ${animDur}s;
    transition-property: color, background;

    &:hover,
    &:focus {
      color: ${linkHoverColor};
    }

    &:active {
      color: ${linkActiveColor};
    }
  }

  ::selection {
    background-color: ${palette.themeAccentLight};
    color: ${palette.white};
  }

  ul,
  ol {
    margin: 0 0 ${paragraphMarginBottom};
    padding: 0 0 0 1em;
  }

  strong {
    font-weight: ${fontWeights.semibold};
  }

  img {
    display: block;
    height: auto;
    width: 100%;
  }
`;

export const RouteFallbackSpinner = styled(Spinner)`
  height: calc(100vh - ${rem(sidebarHeightMobile)});

  @media ${mediaQueries.lg} {
    height: 100vh;
  }
`;
