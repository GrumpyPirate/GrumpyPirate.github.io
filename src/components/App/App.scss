/* ---------------------------------------------------------------------------------------------------------------------
** Global layout styles that apply to all document content
** ------------------------------------------------------------------------------------------------------------------ */
@import '../../sass/functions';
@import '../../sass/variables';
@import '../../sass/mixins';

// Bootstrap
@import '../../sass/bootstrap/bootstrap-custom';

// Vendor library theming
@import '../../sass/vendor';

// Core global
@import '../../sass/type';
@import '../../sass/animations';
@import '../../sass/utility';
@import '../../sass/forms';
@import '../../sass/icons';

.app {
    // lg up
    // @include media-breakpoint-up(lg) {
        display: flex;
        flex-flow: row nowrap;
        align-items: stretch;
    // } // /lg up
} // /.app

.page-header {
    padding: $grid-gutter-width 0;
    margin-bottom: $grid-gutter-width;
    // border-bottom: solid 1px $sidebar-border-color;

    &__title {
        // color: lighten($body-color, 25%);
        color: palette(theme-green-light);
    } // /&__title
} // /.page-header
