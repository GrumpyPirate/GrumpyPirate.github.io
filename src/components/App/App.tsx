import { ApolloProvider } from '@apollo/client';
import React, { FunctionComponent } from 'react';
import { Route, HashRouter as Router, Switch } from 'react-router-dom';
import styled from 'styled-components';

import Footer from 'components/Footer/Footer';
import Main from 'components/Main/Main';
import AboutPage from 'components/Pages/AboutPage/AboutPage';
import HTTPNotFoundPage from 'components/Pages/HTTPNotFoundPage/HTTPNotFoundPage';
import PortfolioItemPage from 'components/Pages/PortfolioItemPage/PortfolioItemPage';
import PortfolioPage from 'components/Pages/PortfolioPage/PortfolioPage';
import Sidebar from 'components/Sidebar/Sidebar';
import contentService from 'services/ContentService';
import { mediaQueries } from 'styles';
import { ClassNameProps } from 'types/common';

import GlobalStyles from './App.constants';

const App: FunctionComponent<ClassNameProps> = ({ className }) => (
  <ApolloProvider client={contentService.client}>
    <GlobalStyles />
    <Router basename="/">
      <div className={className}>
        <Sidebar />
        <Main>
          <Switch>
            <Route exact path="/" component={AboutPage} />
            <Route exact path="/portfolio" component={PortfolioPage} />
            <Route exact path="/portfolio/:slug" component={PortfolioItemPage} />
            <Route exact path="/404" component={HTTPNotFoundPage} />
            <Route component={HTTPNotFoundPage} />
          </Switch>
          <Footer />
        </Main>
      </div>
    </Router>
  </ApolloProvider>
);

export default styled(App)`
  padding-top: $sidebar-height-mobile;

  @media ${mediaQueries.lg} {
    align-items: stretch;
    display: flex;
    flex-flow: row nowrap;
    padding-top: 0;
  }
`;
