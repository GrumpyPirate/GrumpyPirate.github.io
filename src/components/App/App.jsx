// React
import React from 'react'

// Router
import {
    BrowserRouter as Router,
    Route
} from 'react-router-dom'
import Routes from '../../config/routes'

// Components
import Sidebar from '../Sidebar/Sidebar.jsx'

// SCSS
import './App.scss'

// Define App
class App extends React.Component {
    constructor(props) {
        super(props)

        this.state = {}
    } // /constructor(props)

    render() {
        return (
            <Router>
                <div className="app">
                    <Sidebar />
                    <main className="content-container">
                        {Routes.map((route, index) =>
                            <Route
                                key={index}
                                path={route.path}
                                exact={route.exact}
                                component={route.components.content}
                            />
                        )}
                    </main>
                </div>
            </Router>
        )
    } // /render()
} // /class App extends React.Component

export default App
