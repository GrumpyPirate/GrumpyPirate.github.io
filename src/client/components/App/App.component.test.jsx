import React from 'react';
import { shallow } from 'enzyme';

import App from './App.component';

describe('Components', () => {
  describe('App', () => {
    it('should be defined', () => {
      expect(App).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof App).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<App />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
