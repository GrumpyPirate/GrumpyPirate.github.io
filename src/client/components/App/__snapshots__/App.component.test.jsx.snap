// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components App should render as expected, without crashing 1`] = `
<BrowserRouter
  basename="/"
>
  <div
    className="app"
  >
    <withRouter(Connect(Sidebar)) />
    <withRouter(Connect(Main))
      renderLocation={[Function]}
    >
      <Footer />
    </withRouter(Connect(Main))>
  </div>
</BrowserRouter>
`;
